src/menu/charprofile.c

```diff
@@ -0,0 +1,368 @@
+			.pos = { SCREEN_W/2 + 240 + 320 * pofs, SCREEN_H - spr->h * 0.5 },
+			.sprite_ptr = spr,
+			.shader = "sprite_default",
+			.color = RGBA(pbrightness, pbrightness, pbrightness, 1),
+		};
+
+		r_draw_sprite(&portrait_params);
+		portrait_params.sprite_ptr = portrait_get_face_sprite(profiles[selected].name, profiles[selected].faces[ctx->face]);
+		r_draw_sprite(&portrait_params);
+	}
+	r_mat_mv_push();
+	r_mat_mv_translate(SCREEN_W/4, SCREEN_H/3, 0);
+	r_mat_mv_push();
+	r_mat_mv_push();
+
+	if(e->drawdata != 0) {
+		r_mat_mv_translate(0, -300 * e->drawdata, 0);
+		r_mat_mv_rotate(M_PI * e->drawdata, 1, 0, 0);
+	}
+
+	text_draw(profiles[selected].fullname, &(TextParams) {
+		.align = ALIGN_CENTER,
+		.font = "big",
+		.shader = "text_default",
+		.color = RGBA(o, o, o, o),
+	});
+	r_mat_mv_pop();
+
+	if(e->drawdata) {
+		o = 1 - e->drawdata * 3;
+	} else {
+		o = 1;
+	}
+
+	text_draw(profiles[selected].title, &(TextParams) {
+		.align = ALIGN_CENTER,
+		.pos = { 20*(1-o), 30 },
+		.shader = "text_default",
+		.color = RGBA(o, o, o, o),
+	});
+	r_mat_mv_pop();
+
+	r_color4(1, 1, 1, 0.5);
+	text_draw_wrapped(profiles[selected].description, DESCRIPTION_WIDTH, &(TextParams) {
+		.align = ALIGN_LEFT,
+		.pos = { -190, 120 },
+		.font = "small",
+		.shader = "text_default",
+		.color = RGBA(o, o, o, o),
+	});
+	r_mat_mv_pop();
+
+	o = 0.3*sin(m->frames/20.0)+0.5;
+	r_shader("sprite_default");
+
+	r_draw_sprite(&(SpriteParams) {
+		.sprite = "menu/arrow",
+		.pos = { 30, SCREEN_H/3+10 },
+		.color = RGBA(o, o, o, o),
+		.scale = { 0.5, 0.7 },
+	});
+
+	r_draw_sprite(&(SpriteParams) {
+		.sprite = "menu/arrow",
+		.pos = { 30 + 340, SCREEN_H/3+10 },
+		.color = RGBA(o, o, o, o),
+		.scale = { 0.5, 0.7 },
+		.flip.x = true,
+	});
+
+	r_state_pop();
+}
+
+static void action_show_character(MenuData *m, void *arg) {
+	return;
+}
+
+static void add_character(MenuData *m, int i) {
+	if(strcmp(profiles[i].name, "locked")) {
+		log_debug("adding character: %s", profiles[i].name);
+		portrait_preload_base_sprite(profiles[i].name, NULL, RESF_PERMANENT);
+		profiles[i].sprite = portrait_get_base_sprite(profiles[i].name, NULL);
+		MenuEntry *e = add_menu_entry(m, NULL, action_show_character, NULL);
+		e->transition = NULL;
+	}
+}
+
+static void charprofile_free(MenuData *m) {
+	dynarray_foreach_elem(&m->entries, MenuEntry *e, {
+		free(e->arg);
+	});
+}
+
+static bool charprofile_input_handler(SDL_Event *event, void *arg) {
+	MenuData *m = arg;
+	CharProfileContext *ctx = m->context;
+	TaiseiEvent type = TAISEI_EVENT(event->type);
+
+	int prev_cursor = m->cursor;
+
+	if(type == TE_MENU_CURSOR_LEFT) {
+		m->cursor--;
+		ctx->face = 0;
+	} else if(type == TE_MENU_CURSOR_RIGHT) {
+		m->cursor++;
+		ctx->face = 0;
+	} else if(type == TE_MENU_ACCEPT) {
+		// show different expressions for selected character
+		ctx->face++;
+		if(!profiles[m->cursor].faces[ctx->face]) ctx->face = 0;
+	} else if(type == TE_MENU_ABORT) {
+		play_sfx_ui("hit");
+		close_menu(m);
+	}
```

Comment:
Since this function is currently called for every single SDL event, maybe add an `else { return false; }` clause to prevent the code below from being executed unnecessarily.